pages/index.tsx

```diff
@@ -104,11 +102,17 @@ function Home({
   // retrieve query parameter with name cityFilter to set input value
   const { cityFilter } = router.query;
   // created state cityFilterQuery and initialized with query param cityFilter
-  const [cityFilterQuery, setCityFilterQuery] = useState<string>(cityFilter);
+  const [cityFilterQuery, setCityFilterQuery] = useState<string>(() => {
+    if (cityFilter === typeof "string") {
+      return cityFilter;
+    } else {
+      return "";
```

Comment:
If it is not a string (undefined probably) just return empty string: ""

This fixes the Typing problem: city is now always a string with a city like "Alkmaar" or empty string: ""